File: pinyougou-parent/pinyougou-manager-web/src/main/webapp/js/service/specificationService.js
app.service('specificationService', function ($http) {
    this.reloadList = function (page, size, model) {
        return $http.post('/specification/showPageList.do?page=' + page + '&size=' + size, model);
    }
    this.insertSpecificationWithSpecificationOptions = function (model) {
        return $http.post('/specification/insertSpecificationWithSpecificationOptions.do', model);
    }
    this.findByid = function (id) {
        return $http.get('/specification/findByid.do?id=' + id);
    }
    this.updateSpecificationWithSpecificationOptions = function (model) {
        return $http.post('/specification/updateSpecificationWithSpecificationOptions.do', model);
    }
    this.deleteSpecificationWithSpecificationOptionsByids = function (ids) {
        return $http.get('/specification/deleteSpecificationWithSpecificationOptionsByids.do?ids=' + ids);
    }
})

File: pinyougou-parent/pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/SpecificationService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbSpecification;
import com.pinyougou.pojovo.SpecificationWithSpecificationOptions;
import com.pinyougou.result.ResultPage;

import java.util.List;
import java.util.Map;

/**
 * 服务层接口
 *
 * @author Administrator
 */
public interface SpecificationService {

    ResultPage<TbSpecification> showPageList(TbSpecification specification, int page, int size);

    void insertSpecificationWithSpecificationOptions(SpecificationWithSpecificationOptions specificationWithSpecificationOptions);

    SpecificationWithSpecificationOptions findSpecificationWithSpecificationOptionsByid(Long specificationId);

    void updateSpecificationWithSpecificationOptions(SpecificationWithSpecificationOptions specificationWithSpecificationOptions);

    void deleteSpecificationWithSpecificationOptionsByids(Long[] ids);

    List<Map> specificationListForTypeTemplate();
}
